m4/pbs_default_scheduler.m4

```diff
@@ -0,0 +1,51 @@
+
+#
+#  Copyright (C) 1994-2017 Altair Engineering, Inc.
+#  For more information, contact Altair at www.altair.com.
+#   
+#  This file is part of the PBS Professional ("PBS Pro") software.
+#  
+#  Open Source License Information:
+#   
+#  PBS Pro is free software. You can redistribute it and/or modify it under the
+#  terms of the GNU Affero General Public License as published by the Free 
+#  Software Foundation, either version 3 of the License, or (at your option) any 
+#  later version.
+#   
+#  PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY 
+#  WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+#  PARTICULAR PURPOSE.  See the GNU Affero General Public License for more details.
+#   
+#  You should have received a copy of the GNU Affero General Public License along 
+#  with this program.  If not, see <http://www.gnu.org/licenses/>.
+#   
+#  Commercial License Information: 
+#  
+#  The PBS Pro software is licensed under the terms of the GNU Affero General 
+#  Public License agreement ("AGPL"), except where a separate commercial license 
+#  agreement for PBS Pro version 14 or later has been executed in writing with Altair.
+#   
+#  Altair’s dual-license business model allows companies, individuals, and 
+#  organizations to create proprietary derivative works of PBS Pro and distribute 
+#  them - whether embedded or bundled with other software - under a commercial 
+#  license agreement.
+#  
+#  Use of Altair’s trademarks, including but not limited to "PBS™", 
+#  "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's 
+#  trademark licensing policies.
+#
+
+AC_DEFUN([PBS_AC_DEFAULT_SCHEDULER],
+[
+  AC_MSG_CHECKING([for Default scheduler name])
+  AC_ARG_VAR([PBS_DFLT_SCHED_NAME], [Specifies the PBS default scheduler name.])
+  AS_IF([test "x$PBS_DFLT_SCHED_NAME" = "x"],
```

Comment:
@mike0042 
We are actually preserving backward compatibility in the client code(qmgr.c) i.e. in other words when the user does not pass sched name i.e. the case of default scheduler we are automatically passing the name "main" in the client code.
We have made the scheduler name as configurable just in case if the users don't like the name "main" for default scheduler, they can still give any name that they like.  As @mike0042  suggested may be we can change the default scheduler name to "default" instead of "main".